topdown/eval.go

```diff
@@ -2028,8 +2028,9 @@ func (e evalFunc) partialEvalSupportRule(rule *ast.Rule, path ast.Ref) error {
 			}
 
 			e.e.saveSupport.Insert(path, &ast.Rule{
-				Head: head,
-				Body: plugged,
+				Head:        head,
+				Body:        plugged,
+				Annotations: rule.Annotations,
```

Comment:
Is a deep copy needed here 🤔

Reply:
Unless anything modifies either annotations list after-the-fact, I don't think it's necessary. But if we can't be sure, I think a deep copy is preferred.

Reply:
Updated to make a deep copy.